docassemble/MAVirtualCourt/data/questions/209A_affidavit.yml

```diff
@@ -0,0 +1,499 @@
+  mark_unfilled_fields_empty(interview_metadata["complaint_209A_Affidavit"])
+  complaint_209A_Affidavit_preview_question # Pre-canned preview screen
+  basic_questions_signature_flow
+  users[0].signature
+  complaint_209A_Affidavit = True
+---
+continue button field: complaint_209A_Affidavit_preview_question
+question: |
+  Placeholder preview screen
+subquestion: |
+  Here is a preview of the form you'll sign on the next page. 
+  
+   ${complaint_209A_Affidavit_attachment_preview}
+---
+comment: |
+  (KG) Included because we want the name of the defendant. 
+  Should be included from another interview later
+question: |
+  The Defendant
+subquestion: |
+  Tell me about the Defendant.
+fields:
+  - 'Defendant Name': defendant
+---
+question: |
+  Children
+subquestion: |
+  Do you have any children under eighteen?
+fields:
+  - 'Do you have children under the age of eighteen?': plaintiff_has_minor_children
+    datatype: yesnoradio # a yes here should trigger multiple other interviews: page of the 209A complaint (always), care and custody affidavit (always, but maybe not if the parties don't have any kids under 18 TOGETHER. SMEs to check.).
+---
+code: |
+  incidents_of_abuse[i].year
+  incidents_of_abuse[i].time_of_day
+  incidents_of_abuse[i].location
+  incidents_of_abuse[i].initializeAttribute('abuse', DADict)
+  incidents_of_abuse[i].abuse.feared_harm
+  incidents_of_abuse[i].description
+  
+  if not incidents_of_abuse[i].police_included:
+    if not incidents_of_abuse[i].police_involved:
+      if incidents_of_abuse[i].police_why_not:
+        incidents_of_abuse[i].description = incidents_of_abuse[i].police_description
+    else:
+      if incidents_of_abuse[i].add_police:
+        incidents_of_abuse[i].description = incidents_of_abuse[i].police_description
+
+  if plaintiff_has_minor_children:
+    if incidents_of_abuse[i].injuries_self_child:
+      if not incidents_of_abuse[i].injuries_describe:
+        if incidents_of_abuse[i].injuries_edit:
+          incidents_of_abuse[i].injuries_description
+  else:
+    if not incidents_of_abuse[i].injuries:
+      if incidents_of_abuse[i].injuries_edit:
+        incidents_of_abuse[i].description = incidents_of_abuse[i].injuries_description
+    
+  if incidents_of_abuse[i].medical_add:
+    incidents_of_abuse[i].medical_treatment = incidents_of_abuse[i].medical_treatment
+    
+  incidents_of_abuse[i].feelings
+  incidents_of_abuse[i].summary
+  
+  incidents_of_abuse[i].complete = True
+---
+id: 209A Affidavit Date
+question: |
+  % if started:
+    Another incident date
+  % else:
+    Most recent date
+  % endif
+subquestion: |
+  When was the last time ${defendant} abused you?
+  We need to put the year on the form.
+  
+  For the date, try to remember the exact date. If you are not sure of the exact date, do not guess.
+fields:
+  - Year: incidents_of_abuse[i].year
+    code: |
+      reversed(list(range(today().year - 20, today().year + 1)))
+  - Date: incidents_of_abuse[i].date
```

Comment:
If they're not supposed to guess, why is 'Date' a required field?

Reply:
Some kind of date is required. The variable is not a date variable. If plaintiff were to put in an exact day and month it is important that the date is correct, they should not guess at the day and month, but they must put something  to describe the date - as explained in the note on line 207.